custodial-copy-backend/src/main/scala/uk/gov/nationalarchives/custodialcopy/OcflService.scala

```diff
@@ -19,11 +21,15 @@ import scala.jdk.FunctionConverters.*
 
 class OcflService(ocflRepository: MutableOcflRepository, semaphore: Semaphore[IO]) {
 
+  given Logger[IO] = Slf4jLogger.getLogger[IO]
+
+  private def onError(err: Throwable): IO[Unit] = Logger[IO].error(err)(err.getMessage) >> semaphore.release
+
   def commitStagedChanges(id: UUID): IO[Unit] =
     semaphore.acquire >> IO.whenA(ocflRepository.hasStagedChanges(id.toString)) {
-      IO.blocking {
+      IO.blocking[Unit] {
         ocflRepository.commitStagedChanges(id.toString, null)
-      }.void
+      }.onError(onError)
     } >> semaphore.release
```

Comment:
The `onError` method that you created does a `>> semaphore.release`; what happens if you do `>> semaphore.release` again like you have here? Do you need to have them in both places? Same question for the other instances of this below

Reply:
It will only do semaphore.release once. If there's an error it will do it inside onError and ignore the one on line 33 and vice versa if there's no error.

Reply:
Ah, I see, so is the reason why we're do it in the `onError`, directly after logging, in case it doesn't return and hit line 33?